exercises/isogram/src/example.erl

```diff
@@ -0,0 +1,20 @@
+-module(example).
+
+-export([is_isogram/1, test_version/0]).
+
+is_isogram(String) ->
+    check_isogram(string:to_lower(String)).
+
+check_isogram("") ->
+    true;
+check_isogram([H|T]) ->
+    B = alpha(H) andalso found_in(H, T),
+    not B andalso check_isogram(T).
+
+alpha(C_lower) ->
+    C_lower /= string:to_upper(C_lower).
```

Comment:
So `A` is not alpha? Also I like to prepend predicates with `is_`.

As I usually stick to 7 Bit ASCII, the following is sufficient:

```erl
is_alpha(C) when ($A <= C), (C <= $Z) -> true;
is_alpha(C) when ($a <= C), (C <= $z) -> true;
is_alpha(_) -> false.
```

Even though you do not export this function, so you can controll that it is only called with lowercase, it might be found and picked up by search engines and people who just need to check if a char is alphabetic and pick up the only half working function.

Reply:
Yes, like it ... when a language has guards I could use to prevent bad calls, why use just the name of a parameter to suggest there are limitations.

The README for the pangram exercise (effectively) states 7-bit ASCII but not that for the isogram exercise.  Should I change the isogram README ?

I was considering implementing the pangram exercise too.  I propose to finish isogram as is and do pangram using the generator, once you are happy with it and its documentation, as a way of giving it some real world, dumb user, testing.

If you agree I will open a WIP issue for pangram so there is a context in which you can let me know when the generator is ready.

Reply:
> The README for the pangram exercise (effectively) states 7-bit ASCII but not that for the isogram exercise. Should I change the isogram README ?

Sticking to 7Bit ASCII is an unwritten rule in exercism, decided upon about a year ago.

It has not reached all the tracks (or they have extended their tests by will).

But since erlang does not have proper unicode support in 18 and 19, I tend to leave everything as is until 20 is the lower end of the supported OTP versions.

And please do not change the README. It usually is generated via configlet and your changes might get overwritten.

> I was considering implementing the pangram exercise too. I propose to finish isogram as is and do pangram using the generator, once you are happy with it and its documentation, as a way of giving it some real world, dumb user, testing.

I'm happy if you do pangram, generator or not ;)

I probably won't be happy with the generator and its documentation until everything gets generated ;) It just gets extended as I realize I need it.

I'll try to write up a short getting started for the generator until next sunday. We can then add more documentation as you go and ask questions ;)